### DISCUSSION TITLE
question

### DISCUSSION THREAD

**Discussion by PlayGame666** [skyline-emu/skyline#1816] [Q&A]
this question does not apply to the skyline emulator, I just have a question: why, after the wii emulator, did they immediately start making the nintendo switch emulator, skipping the wii U?

**Comment by guranu** [comment]
Because it is easier to make a switch emulator than a wii u emulator.

**Reply by marcemira** [reply]
Correct me if I'm wrong, but I think it's easier because wii u's cpu is a PowerPC based one (https://en.wikipedia.org/wiki/Espresso_(microprocessor)) on which you must emulate/interpret all the instruction, which makes it hard/slow/complicated. But instead, Switch has an ARM 64bits CPU (like most smartphones/tablets, new macs and some other notebooks today) wich, in case of Skyline for Android, uses the same instruction set, means you don't have to emulate the CPU, just execute code, resulting in faster development / straight forward execution. The main issue is the GPU tho, Switche's NVIDIA Tegra is complicated to emulate with current smartphone SOCs GPUs (qualcomm, mali, etc). That's mostly skylines challenge today.

**Reply by guranu** [reply]
Yeah well then then why does the wii u has an amd radeon gpu then?

**Reply by marcemira** [reply]
🤷🏻‍♂️ No idea @guranu... but why not? It was probably cheaper hehe

**Reply by guranu** [reply]
if they used a gpu that phones used back then, than maybe.
